### PR TITLE
cmd/chihaya: add config reloading via SIGUSR1

### LINKED ISSUE #215: support reloading config via unix signal
Given a SIGHUP, I think chihaya should shutdown everything except for the peer store, and then start back up with a newly parsed config.

### PR DESCRIPTION
**Pull request by mrd0ll4r**
I made this a SIGUSR1 instead of a SIGHUP. I know I know, many things use SIGHUP for this kind of stuff, but I think that's because there was nothing else when they were written. We have user-defined signals now, so why not use them?

Fixes #215

### CODE REVIEW

File: cmd/chihaya/main.go [hunk 1]
```diff
@@ -65,67 +64,89 @@ func rootCmdRun(cmd *cobra.Command, args []string) error {
...
+					log.Debug("Successfully restarted")
+					continue
+				}
+			case <-shutdown:
+			}
 
-		if udpFrontend != nil {
-			udpFrontend.Stop()
-		}
+			// Stop frontends
```
Comment (jzelinskie) [inline cmd/chihaya/main.go]: After you reload the config, won't this code still execute and shutdown everything?

Comment (mrd0ll4r) [inline cmd/chihaya/main.go]: That's what all the `continue`s are for - the jump back to the `for`, blocking on the `select` again. (I honestly wasn't sure whether `continue` in a `select` did some other funky stuff, but I tested it and it works :) )

Comment (jzelinskie) [inline cmd/chihaya/main.go]: Why not explicitly put that code in the shutdown case?

Comment (mrd0ll4r) [inline cmd/chihaya/main.go]: We also shut down gracefully on `SIGINT` and `SIGTERM`, i.e. on receiving from the `sig` channel, skipping the `if`, then falling out of the `select`

Comment (jzelinskie) [inline cmd/chihaya/main.go]: I see now. Would it be more clear if we had two signal channels? One dedicated to config reloading and the other to handling termination? Then we can be explicit about the fallthrough to the shutdown logic.

Comment (mrd0ll4r) [inline cmd/chihaya/main.go]: Good idea! ptal

File: cmd/chihaya/main.go [hunk 2]
```diff
@@ -65,67 +64,92 @@ func rootCmdRun(cmd *cobra.Command, args []string) error {
...
-	}
-
-	if cfg.UDPConfig.Addr != "" {
-		udpFrontend = udpfrontend.NewFrontend(logic, cfg.UDPConfig)
 
-		go func() {
-			log.Infoln("started serving UDP on", cfg.UDPConfig.Addr)
-			if err := udpFrontend.ListenAndServe(); err != nil {
-				errChan <- errors.New("failed to cleanly shutdown UDP frontend: " + err.Error())
+			// Stop frontends
```
Comment (jzelinskie) [inline cmd/chihaya/main.go]: I think moving this into the `case` block for _shutdown_ and then explicitly adding `fallthrough` inside the _quit_ case makes the logic more clear.

Comment (mrd0ll4r) [inline cmd/chihaya/main.go]: You cannot `fallthrough` in `select`s: https://play.golang.org/p/TWkkBrxZap. You cannot even `goto` the other case. The clearest solution would be something like this (see commit), but meh. Any suggestions?

**Pull request merged**